src/ur_arm.cpp

```diff
@@ -441,14 +445,13 @@ URArm::KinematicsData URArm::get_kinematics(const ProtoStruct&) {
 }
 
 void URArm::stop(const ProtoStruct&) {
-    if (current_state_->trajectory_running.load()) {
+    if (current_state_->trajectory_status.load() == TrajectoryStatus::k_running) {
         const bool ok = current_state_->driver->writeTrajectoryControlMessage(
             urcl::control::TrajectoryControlMessage::TRAJECTORY_CANCEL, 0, RobotReceiveTimeout::off());
         if (!ok) {
             VIAM_SDK_LOG(warn) << "URArm::stop driver->writeTrajectoryControlMessage returned false";
-            return;
+            throw std::runtime_error("failed to write trajectory control cancel message to URArm");
         }
-        current_state_->trajectory_running.store(false);
```

Comment:
Is the idea that we don't need this `store` anymore because `trajectory_done_cb` ought to be doing that for us?